src/nix/key.rs

```diff
@@ -17,6 +23,18 @@ pub struct Key {
     pub(super) permissions: String,
 }
 
+impl Key {
+    pub(crate) async fn reader(&'_ self,) -> Result<Box<dyn AsyncRead + Send + Unpin + '_>, io::Error> {
```

Comment:
This function seems very unit-test-able, however I see that there are no unit tests as of now. Should I write one?

Reply:
Sure, please do! A lot of the interactions with Nix commands can be unit-tested as well, but I haven't got to write them.